bokeh/sphinxext/bokeh_gallery.py

```diff
@@ -98,19 +98,25 @@ def config_inited_handler(app, config):
     # left at the end (and remove it in that case)
     extras = set(os.listdir(gallery_dir))
 
-    # app.env.note_dependency(specpath)
-    spec = json.load(open(gallery_file))
-    details = spec["details"]
-
-    names = {x["name"] for x in details}
-    if len(names) < len(details):
-        raise SphinxError(f"gallery file {gallery_file!r} has duplicate names")
+    basepath = app.srcdir.replace('/sphinx/source', '')
+    folder = ['plotting','models']
+    details = []
+    detail_names = []
+    for f in folder:
+        path = f'examples/{f}/file'
+        for name in os.listdir(join(basepath, path)):
+            if not name.startswith('_') and name.endswith('.py') and  not 'custom_layout' in name:
+                name = name.replace('.py', '')
+                detail_name = name + '.rst'
+                if detail_name in detail_names:
+                    detail_name = detail_name.replace('.rst', f'_{f}.rst')
+                details.append({"path": join(path,name+'.py'), "name":name, "detail_file_name":detail_name})
```

Comment:
This would be a good place for either a `TypedDict` or a `dataclass`

Reply:
Also probably good to record if this file is for `plotting` or `models`. Yes if we need that later (e.g. for separate indices, or for annotating global index entries) we could parse it out of the file path. But that's both more expensive and potentially fragile. We have the data (as data) available now so we should just record it.

Reply:
At the moment I only add the `models` information if the name is already in the list. Do you want to add this sting to all names? I thought it is easier to read without this extention.

Reply:
What is the benefit of this TypedDict? I added it, but I can not sea why this makes the code better.

Reply:
It can be subject to static type analysis with MyPy which is a good way to avoid many classes of errors before runtime.

Reply:
@mosc9575 I was suggesting recording `models` vs `plotting` in the metadata so that the separate index pages I describe in https://github.com/bokeh/bokeh/pull/12114#discussion_r874212069 could be constructed more simply. O do thing in a "global" index of all examples it might be good to have the location in parens after the name?
```
bar_basic (plotting)
anscombe (models)
```
What do you think?